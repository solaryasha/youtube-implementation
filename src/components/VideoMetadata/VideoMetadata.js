import React from 'react';
import './VideoMetadata.scss';
import { Button, Icon, Divider } from 'semantic-ui-react';
import Rating from '../Rating/Rating';

const VideoMetadata = (props) => {
  const viewCount = Number(props.viewcount).toLocaleString() || '';

  return (
    <div className="video-metadata">
      <h3>Stargazer - A Chillwave Mix</h3>
      <div className="video-stats">
        <span>
          {viewCount}
          {' '}
          views
        </span>
        <div className="video-actions">
          <Rating likes={75} dislikes={25} />
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
          <Button basic icon>
            <Icon name="plus circle" />
          </Button>
          <Button basic icon>
            <Icon name="ellipsis horizontal" />
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
};

export default VideoMetadata;
