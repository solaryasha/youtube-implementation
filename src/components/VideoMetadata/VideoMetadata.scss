@import '../../styles/shared.scss';

.video-metadata {
    color:#606060;
    font-size: 16px;

    h3 {
        color: $text-color-dark;
        font-weight: 400;
        
    }
    .video-stats {
        display: flex;
        justify-content: space-between;
        align-items: center;


        .video-actions{
            display: flex;
            align-items: center;
        }

        & > * :not(:last-child){
            margin-right: 8px;
        }
    }
}