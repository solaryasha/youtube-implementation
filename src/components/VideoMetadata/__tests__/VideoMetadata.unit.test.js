import React from 'react';
import {shallow} from 'enzyme';
import VideoMetadata from '../VideoMetadata';

describe('VideoMetadata', () => {
    test('renders without props', () => {
        const wrapper = shallow(<VideoMetadata />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('renders with view props', () => {
        const wrapper = shallow(<VideoMetadata viewcount={50000} />);
        expect(wrapper).toMatchSnapshot();
    })

})