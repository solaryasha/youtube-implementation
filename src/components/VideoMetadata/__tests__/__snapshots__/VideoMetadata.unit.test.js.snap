// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMetadata renders with view props 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Stargazer - A Chillwave Mix
  </h3>
  <div
    className="video-stats"
  >
    <span>
      50,000
       views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikes={25}
        likes={75}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="plus circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;

exports[`VideoMetadata renders without props 1`] = `
<div
  className="video-metadata"
>
  <h3>
    Stargazer - A Chillwave Mix
  </h3>
  <div
    className="video-stats"
  >
    <span>
      NaN
       views
    </span>
    <div
      className="video-actions"
    >
      <Rating
        dislikes={25}
        likes={75}
      />
      <Button
        as="button"
        basic={true}
        icon={true}
        labelPosition="left"
      >
        <Icon
          as="i"
          name="share"
        />
        Share
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="plus circle"
        />
      </Button>
      <Button
        as="button"
        basic={true}
        icon={true}
      >
        <Icon
          as="i"
          name="ellipsis horizontal"
        />
      </Button>
    </div>
  </div>
  <Divider />
</div>
`;
